### ISSUE TITLE
Popover components (dropdown menu, combobox, etc) don't always close on outside click

### ISSUE THREAD

**Issue by dacobo** [melt-ui/melt-ui#1259]
### Describe the bug

When a popper component is open and an outside button is clicked, there are scenarios where the popper stays open. I've been trying to narrow down what causes the issue, and it seems to be related to the conditional rendering of the button being clicked.

I've noticed it also happens in certain scenarios when clicking another Melt control. In my sample below, you can see this when clicking the Combo Box and then the Dropdown Menu.

### Reproduction

1. Expand a popper control
2. While it's open, click on a button somewhere else on the page
  a. This button must then trigger some change to the state, which then causes said button to be removed from the DOM
3. The popper will stay open

I've put together a sample demonstrating the problem here: https://www.sveltelab.dev/gsi5tthvo4231ts

(Note: It can take a few seconds before the components become responsive in this Sveltelab)

### System Info

```bash
System:
    OS: Windows 11 10.0.22631
    CPU: (12) x64 12th Gen Intel(R) Core(TM) i7-1250U
    Memory: 11.85 GB / 31.63 GB
  Binaries:
    Node: 20.15.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.7.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.22621.3527
  npmPackages:
    @melt-ui/svelte: ^0.83.0 => 0.83.0
    @sveltejs/kit: ^2.0.0 => 2.5.20
    svelte: ^4.2.7 => 4.2.18
```

### Severity

annoyance

**Comment by RLesser** [comment]
This happens for me as well, but specifically when clicking a canvas while the combobox is open. If I click a different non-canvas div it closes as expected. When I click the canvas the input field is losing focus so it's definitely recording the click happened, just not closing the dropdown.

**Comment by fiioonnn** [comment]
Same for me. When i have multiple dropdowns and one of them is open/active, when i click on a differents dropdowns trigger, both dropdowns are open.

**Comment by ybonchev** [comment]
yep have the same exact thing happening, but it only happens in production, in my local environment it works as it should...weird

**Comment by ybonchev** [comment]
for anybody who looks for some sort of solution, you can do next:
1) add some sort of unique id inside your component, i used createId method from @paralleldrive/cuid2, so:
`  import { createId } from '@paralleldrive/cuid2';
  const popoverId = createId();
`
2) then you add a simple store like this:
`export const popoverStore: Writable<string> = writable('');`
3) next you need to add the onOpenChange function in createPopover method, whic in my case looks like this:
`onOpenChange: ({ next }) => {`
      `if (next) {`
        `$popoverStore = popoverId;`
        `open = next;`
      `} else if (!next && $popoverStore === popoverId) {`
        `$popoverStore = '';`
     }
      `return next;`
   }
4) and for the last thing you need to add a callback that will close all popovers on the page:
`  $: if ($popoverStore !== popoverId) {
    open = false;
  }`
simple as that, and it works ;)
